packages/fetch-and-compile/test/fetch.test.ts

```diff
@@ -0,0 +1,118 @@
+import assert from "assert";
+import "mocha";
+import Config from "@truffle/config";
+import { fetchAndCompile } from "../lib/index";
+import axios from "axios";
+import sinon from "sinon";
+const fixture: any = require("./fixture.js");
+
+function stubAxiosGetMethod(url: string, address: string, data: object) {
+  sinon
+    .stub(axios, "get")
+    .withArgs(url, {
+      params: {
+        module: "contract",
+        action: "getsourcecode",
+        address: address,
+        apikey: ""
+      },
+      responseType: "json",
+      maxRedirects: 50
+    })
+    .returns(Promise.resolve({ data: data }));
+}
+
+afterEach(function () {
+  //TS can't detect that is a sinon stub so we have to use ts-ignore
+  //@ts-ignore
+  //restoring stub
+  axios.get.restore();
+});
+
+describe("fetchAndCompile", function () {
+  it("verifes contract from mainnet", async function () {
+    const config = Config.default().merge({
+      networks: {
+        mainnet: {
+          network_id: 1
+        }
+      },
+      network: "mainnet"
+    });
+    const address = "0x7a250d5630B4cF539739dF2C5dAcb4c659F2488D";
+    stubAxiosGetMethod(
+      "https://api.etherscan.io/api",
+      address,
+      fixture.mainnetData
+    );
+    //TS can't detect that is a sinon stub so we have to use ts-ignore
+    //@ts-ignore
+    axios.get.callThrough();
+    const result = await fetchAndCompile(address, config);
+    let contractNameSourceInfo = result.sourceInfo.contractName;
+    let contractNameCompliations =
```

Comment:
A few notes here.  In order of least important to most important:

1. This `let` can be a `const`.
2. This variable name is misspelled; you have "compliations" instead of "compilations".
3. But most importantly, this variable name is misleading; you need to rename this.

Above, you have a variable named `contractNameSourceInfo`.  That's a little bit confusing, but context makes it clear that you mean the contract name coming from the source info.

But then you have a second variable called `contractNameCompilations`.  So that's the contract name coming from the compilations, right?  Nope!  It's not a contract name at all.  That is therefore a misleading variable name; both it's misleading on its own, and it's made more misleading by the false parallelism.

This variable needs to be given a name that more accurately reflects what it is.

Reply:
Commented on this below. Hope the new names are a bit less confusing :)